server/services/vasRequest.service.js

```diff
@@ -81,11 +81,40 @@ export const verifyVasRequestInput = async (vasDetails) => {
   }
 };
 
+export const similarVasRequestExists = async (vasDetails) => {
+  let searchparameters = {
+    userId: ObjectId(vasDetails.userId),
+    vasId: ObjectId(vasDetails.vasId),
```

Comment:
I believe the current implementation can be improved. I think you should be able to add the conditional field members here:
```suggestion
    vasId: ObjectId(vasDetails.vasId),
    ...(vasDetails.propertyId && {propertyId: ObjectId(vasDetails.propertyId)}),
    ...(vasDetails.offerId && {offerId: ObjectId(vasDetails.offerId)}),
```

**Source**
https://stackoverflow.com/questions/11704267/in-javascript-how-to-conditionally-add-a-member-to-an-object/40560953#40560953

--


Another way is to add the 4 fields in an array and add them to the `searchParameter` if a value exists for each field.

```js
const fields = ['userId', 'vasId', 'offerId', 'propertyId'];
// loop through fields, checking if vasDetails[field] exists and then assign to searchParameter object.
```